### DISCUSSION TITLE
How can the body-transformer plugin access query parameters?

### DISCUSSION THREAD

**Discussion by andersonbui** [apache/apisix#11618] [General]
How can the body-transformer plugin access query parameters?

Using the body-transformer plugin, I need to access a piece of data (123456 and AppPrueba001) that comes in the URL of the original request, to send it in the body of the resulting request.

original request:
```bash
curl --location 'https://dominio_apisix.com/v1/123456/messages?appname=AppPrueba001' -H "Host: example.com" \
--data '{
    "to": "6543221",
    "type": "text",
    "text":  "text-message-content"
}'
```

Transformed request:
```bash
curl --location 'https://destinationdomain.com/v1/send_messages?' -H "Host: example.com" \ 
--data '{
    "to": "6543221",
    "type": "text",
    "text":  "text-message-content",
    "from": "123456",
    "appname": "AppPrueba001"
}'
```

How would this be done? Can you help me please?

**Comment by WrightKD** [comment]
Hi @andersonbui 

You can reference a query parameter in a response template using `_ctx.var.arg_*`  where `*` is the name of your query parameter. e.g. `_ctx.var.arg_appname`
